packages/types/src/types/provider/openshift/TypedResource.ts

```diff
@@ -0,0 +1,6 @@
+import { OpenshiftResource } from './Resource';
+
+export interface TypedOpenshiftResource extends OpenshiftResource {
+  // prop added by the UI to implement narrowing (discriminated union)
+  providerType?: 'openshift';
```

Comment:
```suggestion
  providerType: 'openshift';
```

if we want to use ts automation to get the type, we can't use optional fields

Reply:
the prop is populated only for VMs - via dedicated hook. Other types extending TypedResource will not have this prop.

Reply:
changed the prop to required - note that it's not present in all the objects except VMs where (as stated above) it's added via a dedicated hook.